samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -29,20 +44,71 @@ static const struct bt_data ad[] = {
 			  0x12, 0x18,	/* HID Service */
 #endif
 			  0x0f, 0x18),	/* Battery Service */
-	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
-};
 
-static const struct bt_data sd[] = {
+#if CONFIG_DESKTOP_BLE_SWIFT_PAIR
+	BT_DATA(BT_DATA_NAME_SHORTENED, DEVICE_NAME, DEVICE_NAME_SHORT_LEN),
+	BT_DATA_BYTES(BT_DATA_MANUFACTURER_DATA,
+			  0x06, 0x00,	/* Microsoft Vendor ID */
+			  0x03,		/* Microsoft Beacon ID */
+			  0x00,		/* Microsoft Beacon Sub Scenario */
+			  0x80),	/* Reserved RSSI Byte */
+#else
 	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+#endif
 };
 
+static const struct bt_data sd[] = {};
```

Comment:
`BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN)` was removed from scan data and it is needed there for central to correctly use its filters

Reply:
using flag `BT_LE_ADV_OPT_USE_NAME` tells the stack to insert `CONFIG_BT_DEV_NAME` into scan response data.